### PR TITLE
Tactical reload fixes

### PR DESCRIPTION
**Pull request by Duckbuster**
Add the copy pasta AllowReload to missing classes.
(And make them all the same ScrN wide, so you can reload after buying weapon and get the MagCapacity+1)

and

Fix the SP Tommygun to behave like all the other Tommyguns.
(Close the bolt if holding down fire after firing the last bullet)

### CODE REVIEW

File: Classes/ScrnAK47AssaultRifle.uc [hunk 1]
```diff
@@ -6,6 +6,28 @@ var         float             ReloadShortRate;
 
 var transient bool  bShortReload;
 
+// copy-pasted to add (MagCapacity+1)
+simulated function bool AllowReload()
```
Comment (poosh) [inline Classes/ScrnAK47AssaultRifle.uc:10]: I suggest the newer and more efficient code (introduced in pistols, e.g ScrnDeagle.uc):
```
simulated function bool AllowReload()
{
    UpdateMagCapacity(Instigator.PlayerReplicationInfo);

    if( !Other.IsHumanControlled() ) {
        return !bIsReloading && MagAmmoRemaining <= MagCapacity && AmmoAmount(0) > MagAmmoRemaining;
    }

    return !( FireMode[0].IsFiring() || FireMode[1].IsFiring() || bIsReloading || ClientState == WS_BringUp
            || MagAmmoRemaining >= MagCapacity + 1 || AmmoAmount(0) <= MagAmmoRemaining
            || (FireMode[0].NextFireTime - Level.TimeSeconds) > 0.1 );
}
```

File: Classes/ScrnSPThompsonFire.uc [hunk 2]
```diff
@@ -66,6 +66,14 @@ state FireLoop
 
         NextFireTime = Level.TimeSeconds - 0.000001; //fire now!
     }
+    function ModeTick(float dt)
```
Comment (poosh) [inline Classes/ScrnSPThompsonFire.uc:69]: Do we need to do it every tick? Isn't it enough to do it at the `EndState()`? Please check.

**Review by poosh — CHANGES_REQUESTED** [review]
Use the provided AllowReload() code.

File: Classes/ScrnSPThompsonFire.uc (hunk 2 above)
Comment (Duckbuster) [reply Classes/ScrnSPThompsonFire.uc:69]: Good question, I have no idea actually, I just copied what was in the other TommyGuns:
(From ScrnThompsonIncFire.uc and ScrnThompsonDrumFire.uc)

```
// fixes double shot bug -- PooSH
state FireLoop
{
    function BeginState()
    {
        super.BeginState();

        NextFireTime = Level.TimeSeconds - 0.000001; //fire now!
    }
    function ModeTick(float dt)
    {
        if( KFWeap.MagAmmoRemaining < 1 )
        {
            DoCloseBolt(); //plays sound and sets bBoltClosed
        }
        Super.ModeTick(dt);
    }
}
```

File: Classes/ScrnAK47AssaultRifle.uc (hunk 1 above)
Comment (Duckbuster) [reply Classes/ScrnAK47AssaultRifle.uc:10]: Ah, ok, I will replace it with the newer code :) Mine was based off the one in ScrNWeaponPack, will do change there too ;)

**Comment by Duckbuster** [comment]
I updated them to the newer one you suggested.
I noticed some had "bBotControlled" so I replaced those occurrences with "!Other.IsHumanControlled()" like from your example.
I don't know which one is faster, but if needed I can replace them all to bBotControlled.

**Comment by poosh** [comment]
Calling a function is slower than checking a boolean value, but those are microseconds or maybe even nanoseconds that we are talking about. It doesn't matter in the reload function since reloads are not occurring that often. The reason why I introduced `bBotControlled` in pistol code is to use it inside the `Tick()` function.

File: Classes/ScrnDeagle.uc [hunk 3]
```diff
@@ -350,7 +350,7 @@ simulated function bool AllowReload()
 {
     UpdateMagCapacity(Instigator.PlayerReplicationInfo);
 
-    if( bBotControlled ) {
+    if( !Other.IsHumanControlled() ) {
```
Comment (poosh) [inline Classes/ScrnDeagle.uc:353]: Why did you replace this one?

**Review by poosh — APPROVED** [review]
Good work! To be honest, that inconsistency pissed me off, but I was too lazy to fix it all over the code. I'm glad that somebody did it for me (i.e. - you) ;)

File: Classes/ScrnDeagle.uc (hunk 3 above)
Comment (Duckbuster) [reply Classes/ScrnDeagle.uc:353]: Because at this point I didn't know about the bBotControlled used for the pistols :(

**Pull request merged**